### PR TITLE
Implement IniGetReturnTypeExtension

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/3063

required for https://github.com/phpstan/phpstan-src/pull/2358#discussion_r1174357092

### CODE REVIEW

File: src/Type/Php/IniGetReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			'memory_limit' => new StringType(),
+			'max_execution_time' => $numericString,
+			'max_input_time' => $numericString,
+		];
+
+		$argType = $scope->getType($args[0]->value);
+		$results = [];
+		foreach ($argType->getConstantStrings() as $constantString) {
+			if (!array_key_exists($constantString->getValue(), $types)) {
+				continue;
```
Comment (ondrejmirtes) [inline src/Type/Php/IniGetReturnTypeExtension.php]: So what's the default return type? If we encounter `'unknown'|'max_execution_time'`, the return type from this extension will be wrong.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: src/Type/Php/IniGetReturnTypeExtension.php (hunk 1 above)
Comment (staabm) [reply src/Type/Php/IniGetReturnTypeExtension.php]: good catch

**Comment by ondrejmirtes** [comment]
Thank you.

**Comment by ondrejmirtes** [comment]
Although I don't understand how it's gonna help https://github.com/phpstan/phpstan-src/pull/2358#discussion_r1174357092 because for that key it's still gonna be `string|false` AFAIK.

**Comment by staabm** [comment]
I think `precision` can be added to the type list to make it return a `numeric-string`:

```
$ php -r 'var_dump(ini_get("precision"));'
string(2) "14"

$  php -n -r 'var_dump(ini_get("precision"));'
string(2) "14"
```

but thats something I would expect to be part of https://github.com/phpstan/phpstan-src/pull/2358

see also https://www.php.net/manual/en/ini.core.php#ini.precision

**Pull request merged**